src/services/unique_id/UniqueIDGen_service.cc

```diff
@@ -0,0 +1,19 @@
+#include "services/unique_id/UniqueIDGen_service.h"
+
+#include <JANA/JApplication.h>
+#include <map>
+#include <string>
+#include <string_view>
+#include <variant>
+
+UniqueIDGen_service::UniqueIDGen_service(JApplication* app)
+    : m_app(app), m_uniqueGenIDSvc(algorithms::UniqueIDGenSvc::instance()) {
+  for (const auto& [key, prop] : m_uniqueGenIDSvc.getProperties()) {
+    std::visit(
+        [this, key = key](auto&& val) {
+          m_app->SetDefaultParameter(std::string(key), val); // FIXME add description
+          this->m_uniqueGenIDSvc.setProperty(key, val);
+        },
+        prop.get());
+  }
```

Comment:
The `UniqueIDGen_service` implements a JANA2 service that is able to
1. Initialize algorithms-style service (`UniqueIDGenSvc`)
2. Capture a `JApplication*` and use it to initialize parameters

There is already such a JANA2 service in EICrecon that does 1. it's `AlgorithmsInit_service` itself. We can move responsibility 2. there and get rid of this extra indirection. What do you think?

Reply:
Oh, that's a fair point. And I agree, we better stick with the current ongoing approach of shoving everything in `AlgorithmsInit_service`, and refactor uniformly later.

Reply:
Done in 00b7f2d5e.